src/nixpacks/app.rs

```diff
@@ -117,4 +177,51 @@ mod tests {
         );
         Ok(())
     }
+
+    #[test]
+    fn test_find_files() -> Result<()> {
+        let app = App::new("./examples/monorepo")?;
+        let m = app.find_files("**/*.tsx").unwrap();
+        let dir = env::current_dir().unwrap();
+        assert_eq!(
+            m,
+            vec![
+                dir.join("examples/monorepo/packages/client/pages/_app.tsx")
+                    .to_str()
+                    .unwrap(),
+                dir.join("examples/monorepo/packages/client/pages/index.tsx")
+                    .to_str()
+                    .unwrap(),
+            ]
+        );
+        Ok(())
+    }
+
+    #[test]
+    fn test_find_match() -> Result<()> {
+        let app = App::new("./examples/monorepo")?;
+        let re = Regex::new(r"className")?;
+        let m = app.find_match(&re, "**/*.tsx").unwrap();
+        assert!(m);
+        Ok(())
+    }
+
+    #[test]
+    fn test_strip_source_path() -> Result<()> {
+        let app = App::new("./examples/npm")?;
+        let path_to_strip = app.source.join("foo/bar.txt");
+        assert_eq!(
+            app.strip_source_path(path_to_strip.to_str().unwrap())
+                .unwrap(),
+            "foo/bar.txt"
+        );
+        Ok(())
+    }
+
+    #[test]
+    fn test_strip_source_path_no_source_prefix() -> Result<()> {
+        let app = App::new("./examples/npm")?;
+        assert_eq!(app.strip_source_path("no/prefix.txt")?, "no/prefix.txt");
+        Ok(())
+    }
```

Comment:
I might be missing something, but what is the difference between these two tests?

Reply:
Lol, one of them uses a path that's joined onto `app.source`. I wanted to test this because `PathBuf.strip_prefix` errors if the path does not contain the prefix.